plugins/Observability/src/Observability.cpp

```diff
@@ -644,212 +616,19 @@ void Observability::draw(StelCore* core)
 			}
 		}
 
-// Determine source observability (only if something changed):
-		if ((souChanged || locChanged || yearChanged))
-		{
-			lineBestNight.clear();
-			lineObservableRange.clear();
-
-			// Check if the target cannot be seen.
-			if (culmAlt >= (halfpi - refractedHorizonAlt))
-			{
-				//ObsRange = q_("Source is not observable.");
-				//AcroCos = q_("No Acronychal nor Cosmical rise/set.");
-				lineObservableRange = msgSrcNotObs;
-				lineAcro = msgNoAcroRise;
-				lineCosm = msgNoCosmRise;
-				lineHeli = msgNoHeliRise;
-			}
-			else
-			{ // Source can be seen.
-///////////////////////////
-// - Part 1. Determine the best observing night (i.e., opposition to the Sun):
-				if (show_Best_Night)
-				{
-					int selday = 0;
-					double deltaPhs = -1.0; // Initial dummy value
-					double tempPhs; 	
-					for (int i=0; i<nDays; i++) // Maximize the Sun-object separation.
-					{
-						tempPhs = Lambda(objectRA[i], objectDec[i],
-						                 sunRA[i], sunDec[i]);
-						if (tempPhs > deltaPhs)
-						{
-							selday = i;
-							deltaPhs = tempPhs;
-						}
-					}
+      // Determine source observability (only if something changed):
```

Comment:
The hunk below this comment was turned into the `getObjectObservability()` method.